src/mqtt/publishing/publisher.cr

```diff
@@ -32,9 +33,20 @@ module PlaceOS::MQTT
       State.new(key, payload)
     end
 
+    private struct Event
+      include JSON::Serializable
+      @[JSON::Field(converter: Time::EpochConverter)]
+      getter time : Time
+      @[JSON::Field(converter: String::RawConverter)]
```

Comment:
Nice. You can also use `emit_null: true` on here and make it nilable if that doesn’t break things elsewhere too.